## *********************************************************
## 
## File autogenerated for the control_system package 
## by the dynamic_reconfigure package.
## Please do not edit.
## 
## ********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 245, 'name': 'Default', 'parent': 0, 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 273, 'description': 'offset value', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'offset', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': 0.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for pos_x', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_px', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for pos_y', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_py', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for pos_z', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_pz', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for pos_x', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_px', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for pos_y', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_py', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for pos_z', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_pz', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for vel_x', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_vx', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for vel_y', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_vy', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for vel_z', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_vz', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for acc_x', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_ax', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for acc_y', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_ay', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for acc_z', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_az', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for pos_x', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_vx', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for pos_y', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_vy', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for pos_z', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_vz', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for roll', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_roll', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for pitch', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_pitch', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for yaw', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_yaw', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for roll', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_roll', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for pitch', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_pitch', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for yaw', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_yaw', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for vel_roll', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_vel_roll', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for vel_pitch', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_vel_pitch', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for vel_yaw', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_vel_yaw', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for vel_roll', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_vel_roll', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for vel_pitch', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_vel_pitch', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain_I for vel_yaw', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_i_vel_yaw', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for acc_roll', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_acc_roll', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for acc_pitch', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_acc_pitch', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}, {'srcline': 273, 'description': 'Gain for acc_yaw', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/kinetic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Gain_acc_yaw', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -200.0, 'type': 'double'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])    
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

